patch/Linking.js

```diff
@@ -0,0 +1,341 @@
+import Constants from 'expo-constants';
+import { Platform, UnavailabilityError } from 'expo-modules-core';
+import invariant from 'invariant';
+import qs from 'qs';
+import { useEffect, useState } from 'react';
+import URL from 'url-parse';
+import NativeLinking from './ExpoLinking';
+import { hasCustomScheme, resolveScheme } from './Schemes';
+function validateURL(url) {
+    invariant(typeof url === 'string', 'Invalid URL: should be a string. Was: ' + url);
+    invariant(url, 'Invalid URL: cannot be empty');
+}
+function getHostUri() {
+    if (Constants.manifest?.hostUri) {
+        return Constants.manifest.hostUri;
+    }
+    else if (Constants.manifest2?.extra?.expoClient?.hostUri) {
+        return Constants.manifest2.extra.expoClient.hostUri;
+    }
+    else if (!hasCustomScheme()) {
+        // we're probably not using up-to-date xdl, so just fake it for now
+        // we have to remove the /--/ on the end since this will be inserted again later
+        return removeScheme(Constants.linkingUri).replace(/\/--($|\/.*$)/, '');
+    }
+    else {
+        return null;
+    }
+}
+function isExpoHosted() {
+    const hostUri = getHostUri();
+    return !!(hostUri &&
+        (/^(.*\.)?(expo\.io|exp\.host|exp\.direct|expo\.test|expo\.dev)(:.*)?(\/.*)?$/.test(hostUri) ||
```

Comment:
`|expo\.dev` was missing from this regex 
🙃 Keith from expo's discord channel gave me the tip!